File: Fugazi/src/org/fugazi/ast/literal/STRING.java
package org.fugazi.ast.literal;

public class STRING extends Literal {

    private final String value;

    public STRING(String _value) {
        this.value = _value;
    }

    public String getValue() {
        return this.value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    public <T> T accept(ILiteralVisitor<T> visitor) {
        return visitor.visitSTRING(this);
    }
}

File: FelipezConde/QuestionnaireLanguage/Evaluation/Values/String.cs

namespace Evaluation.Values
{
    public class String : Value
    {
        private string value;
        public String(string value)
        {
            this.value = value;
        }
        public override string ToString()
        {
            return "string";
        }

        public string GetValue()
        {
            return value;
        }

        #region Equal
        public override Value Equal(Value value)
        {
            return value.StringEqual(this);
        }
        public override Value StringEqual(Values.String stringValue)
        {
            return new Bool(GetValue().Equals(stringValue.GetValue()));
        }
        #endregion

        #region NotEqual
        public override Value NotEqual(Value value)
        {
            return value.StringNotEqual(this);
        }
        public override Value StringNotEqual(Values.String stringValue)
        {
            return new Bool(!GetValue().Equals(stringValue.GetValue()));
        }
        #endregion

        public override T Accept<T>(IValueVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }
    }
}
